labelbox/data/serialization/ndjson/converter.py

```diff
@@ -40,8 +42,45 @@ def serialize(
         Returns:
             A generator for accessing the ndjson representation of the data
         """
+        used_annotation_uuids = set()
+        for label in labels:
+            annotation_uuid_lookup = {}
+            # UUIDs are private properties used to enhance UX when defining relationships.
+            # They are created for all annotations, but only utilized for relationships.
+            # To avoid overwriting, UUIDs must be unique across labels.
+            # Non-relationship annotation UUIDs are dropped (server-side generation will occur).
+            # For relationship annotations, new UUIDs are generated and stored in a lookup table.
+            for annotation in label.annotations:
+                if isinstance(annotation, RelationshipAnnotation):
+                    source_uuid = annotation.value.source._uuid
+                    target_uuid = annotation.value.target._uuid
+
+                    if (len(
+                            used_annotation_uuids.intersection(
+                                {source_uuid, target_uuid})) > 0):
+                        new_source_uuid = uuid.uuid4()
+                        new_target_uuid = uuid.uuid4()
+
+                        annotation_uuid_lookup[source_uuid] = new_source_uuid
+                        annotation_uuid_lookup[target_uuid] = new_target_uuid
+                        annotation.value.source._uuid = new_source_uuid
+                        annotation.value.target._uuid = new_target_uuid
+
+            for annotation in label.annotations:
+                if (not isinstance(annotation, RelationshipAnnotation) and
+                        "_uuid" in annotation):
+                    annotation._uuid = annotation_uuid_lookup.get(
+                        annotation._uuid, None)
+
         for example in NDLabel.from_common(labels):
-            res = example.dict(by_alias=True)
+            annotation_uuid = getattr(example, "uuid", None)
+
+            annotation_uuid_is_none = annotation_uuid == "None"
+            if not annotation_uuid_is_none:
+                used_annotation_uuids.add(annotation_uuid)
```

Comment:
I must be missing something here regarding `used_annotation_uuids`. We update them here, line 80. However we check them for intersection BEFORE, at line 59... wouldn't it be always empty there?